grype/matcher/common/distro_matchers.go

```diff
@@ -32,7 +35,12 @@ func FindMatchesByPackageDistro(store vulnerability.ProviderByDistro, d *distro.
 		// if the constraint it met, then the given package has the vulnerability
 		isPackageVulnerable, err := vuln.Constraint.Satisfied(verObj)
 		if err != nil {
-			return nil, fmt.Errorf("distro matcher failed to check constraint='%s' version='%s': %w", vuln.Constraint, verObj, err)
+			var e *version.NonFatalConstraintError
+			if errors.As(err, &e) {
+				log.Warn(e.Error())
```

Comment:
nit: Typically in Go, you don't need to call `.Error()` on errors. When passing errors to functions that take `interface{}`, like for formatting strings, `.Error()` will be called automatically if needed. So you can just pass the error itself (`e` in this case).